### PR TITLE
feat(lang): localize job names

### PR DESCRIPTION
**Pull request by TheCatLady**
#### Description

Instead of displaying a hardcoded job name, use a localized job name when available.

#### Screenshot (if UI-related)

N/A

#### To-Dos

- [x] Successful build `yarn build`
- [x] Translation keys `yarn i18n:extract`

#### Issues Fixed or Closed

N/A

### CODE REVIEW

File: overseerr-api.yml [hunk 1]
```diff
@@ -2087,9 +2087,6 @@ paths:
                     id:
                       type: string
                       example: job-name
-                    name:
-                      type: string
-                      example: A Job Name
```
Comment (sct) [inline overseerr-api.yml:2092]: No reason to remove this. It's also a breaking API change. Let's just leave it in.

File: src/components/Settings/SettingsJobsCache/index.tsx [hunk 2]
```diff
@@ -66,7 +77,7 @@ const SettingsJobs: React.FC = () => {
     await axios.post(`/api/v1/settings/jobs/${job.id}/run`);
     addToast(
       intl.formatMessage(messages.jobstarted, {
-        jobname: job.name,
+        jobname: intl.formatMessage(messages[job.id]),
```
Comment (sct) [inline src/components/Settings/SettingsJobsCache/index.tsx:80]: We should design this to check if the message actually exists. If it doesn't exist there should be a fallback string like `Unknown Job.`

File: overseerr-api.yml (hunk 1 above)
Comment (TheCatLady) [reply overseerr-api.yml:2092]: I didn't think it was being used anywhere, but that's a good point.  I'll add it back!

Comment (sct) [reply overseerr-api.yml:2092]: Other people might be using it though. Never know.

Comment (TheCatLady) [reply overseerr-api.yml:2092]: @sct Yeah you're totally right :)

File: server/job/schedule.ts [hunk 3]
```diff
@@ -7,9 +7,9 @@ import downloadTracker from '../lib/downloadtracker';
 
 interface ScheduledJob {
   id: string;
-  job: schedule.Job;
   name: string;
   type: 'process' | 'command';
+  job: schedule.Job;
```
Comment (sct) [inline server/job/schedule.ts:12]: Don't really need this change?

File: src/components/Settings/SettingsJobsCache/index.tsx [hunk 4]
```diff
@@ -11,7 +16,7 @@ import Badge from '../../Common/Badge';
 import { CacheItem } from '../../../../server/interfaces/api/settingsInterfaces';
 import { formatBytes } from '../../../utils/numberHelpers';
 
-const messages = defineMessages({
+const messages: { [messageName: string]: MessageDescriptor } = defineMessages({
```
Comment (sct) [inline src/components/Settings/SettingsJobsCache/index.tsx:19]: Why are we typing this ourselves? `defineMessages` should already be returning the correct type?

File: server/job/schedule.ts (hunk 3 above)
Comment (TheCatLady) [reply server/job/schedule.ts:12]: Oh whoops, that was unintentional!

File: src/components/Settings/SettingsJobsCache/index.tsx (hunk 4 above)
Comment (TheCatLady) [reply src/components/Settings/SettingsJobsCache/index.tsx:19]: It doesn't like it when you do `messages[someString]` :(

**Review by sct — APPROVED** [review]

**Pull request merged**